src/main/java/frc/robot/RobotContainer.java

```diff
@@ -204,17 +214,28 @@ private void configureBindings() {
 
     turretCommandFactory.setDefaultCommand(turretCommandFactory.idleTurret());
 
+    TunableNumber turretTestPitch =
+        new TunableNumber("Turret Test Pitch", 0d, (a) -> {}, turretSubsystem);
+    OI.getTrigger(OI.Driver.lockAmp)
+        .whileTrue(
+            drivetrainCommandFactory.pointInDirection(
+                FieldConstants.AMP_DIRECTION, SwerveSubsystem.scrubRotation(input)));
+    OI.getButton(OI.Driver.lockSource)
+        .whileTrue(drivetrainCommandFactory.pointDrive(robotStateManager.getSourceAngle(), input));
+    OI.getTrigger(OI.Driver.lockSpeaker)
+        .whileTrue(drivetrainCommandFactory.pointDrive(robotStateManager.getSpeakerAngle(), input));
+
+    OI.getButton(OI.Operator.retractClimber)
+        .whileTrue(turretCommandFactory.testTurretCommand(turretTestPitch));
+
     OI.getButton(OI.Driver.resetRotationButton)
         .onTrue(drivetrainCommandFactory.zeroDriveTrain().withName("Put Pose & Rotation on Field"));
 
-    OI.getButton(OI.Driver.useRod)
-        .whileTrue(drivetrainCommandFactory.assistedDriving(input, robotStateManager));
-
     OI.getTrigger(OI.Operator.prepareToFire)
         .whileTrue(
             Commands.either(
                 trapElvCommandFactory.positionAMP(),
-                prepareToScoreSpeaker(),
+                Commands.parallel(prepareToScoreSpeaker()),
```

Comment:
Should this get a `withName()`?

Reply:
no, all the components of this command are with the name